dpe/src/commands/certify_key.rs

```diff
@@ -329,4 +402,214 @@ mod tests {
+                .unwrap();
+        // parse SignedData
+        let mut signed_data =
+            SignedData::from_der(&content_info.content.to_der().unwrap()).unwrap();
+        assert_eq!(signed_data.version, CmsVersion::V1);
+
+        // optional field certificates is not populated
+        assert!(signed_data.certificates.is_none());
+        // optional field crls is not populated
+        assert!(signed_data.crls.is_none());
+
+        // validate hash algorithm OID
+        assert_eq!(signed_data.digest_algorithms.len(), 1);
+        let digest_alg = &signed_data.digest_algorithms.get(0).unwrap();
+        let hash_alg_oid = match DPE_PROFILE.alg_len() {
+            AlgLen::Bit256 => "2.16.840.1.101.3.4.2.1",
+            AlgLen::Bit384 => "2.16.840.1.101.3.4.2.2",
+        };
+        assert!(digest_alg
+            .assert_algorithm_oid(ObjectIdentifier::new_unwrap(hash_alg_oid))
+            .is_ok());
+
+        // validate signer infos
+        let signer_infos = signed_data.signer_infos.0;
+        // ensure there is only 1 signer info
+        assert_eq!(signer_infos.len(), 1);
+        let signer_info = signer_infos.get(0).unwrap();
+        assert_eq!(signer_info.version, CmsVersion::V1);
+
+        // optional field signed_attrs is not populated
+        assert!(signer_info.signed_attrs.is_none());
+
+        // optional field unsigned_attrs is not populated
+        assert!(signer_info.unsigned_attrs.is_none());
+
+        // validate hash algorithm OID
+        assert!(signer_info
+            .digest_alg
+            .assert_algorithm_oid(ObjectIdentifier::new_unwrap(hash_alg_oid))
+            .is_ok());
+
+        // validate signature algorithm OID
+        let sig_alg_oid = match DPE_PROFILE.alg_len() {
+            AlgLen::Bit256 => "1.2.840.10045.4.3.2",
+            AlgLen::Bit384 => "1.2.840.10045.4.3.3",
+        };
+        assert!(signer_info
+            .signature_algorithm
+            .assert_algorithm_oid(ObjectIdentifier::new_unwrap(sig_alg_oid))
+            .is_ok());
+
+        // validate signer identifier
+        let sid = &signer_info.sid;
+        let mut subj_serial = [0u8; DPE_PROFILE.get_hash_size() * 2];
+        let pub_key = EcdsaPub {
+            x: CryptoBuf::new(&certify_resp.derived_pubkey_x).unwrap(),
+            y: CryptoBuf::new(&certify_resp.derived_pubkey_y).unwrap(),
+        };
+        env.crypto
+            .get_pubkey_serial(DPE_PROFILE.alg_len(), &pub_key, &mut subj_serial)
+            .unwrap();
+        match sid {
+            SignerIdentifier::IssuerAndSerialNumber(issuer_and_serial_number) => {
+                let cert_serial_number = &issuer_and_serial_number.serial_number;
+                assert_eq!(&subj_serial[..20], cert_serial_number.as_bytes());
+                // note: the cms crate does not let us convert issuer_and_serial_number.issuer
+                // into bytes so we unfortunately cannot test the issuer name here.
```

Comment:
I don't quite understand. Isn't there an as_bytes function here? https://docs.rs/x509-cert/0.2.3/x509_cert/serial_number/struct.SerialNumber.html#method.as_bytes

Reply:
I am using the cms crate to parse the wrapping ContentInfo  - https://docs.rs/cms/0.2.2/cms/cert/struct.IssuerAndSerialNumber.html

Reply:
i was able to use Name::to_der to parse it actually.